syntax = "proto2";

package object_detection.protos;

import "object_detection/protos/center_net.proto";
import "object_detection/protos/faster_rcnn.proto";
import "object_detection/protos/ssd.proto";
import "object_detection/protos/detr.proto";
import "object_detection/protos/anchor_generator.proto";
import "object_detection/protos/box_predictor.proto";
import "object_detection/protos/hyperparams.proto";
import "object_detection/protos/image_resizer.proto";
import "object_detection/protos/losses.proto";
import "object_detection/protos/post_processing.proto";

// Top level configuration for DetectionModels.
message DetectionModel {
  oneof model {
    FasterRcnn faster_rcnn = 1;
    Ssd ssd = 2;

    // This can be used to define experimental models. To define your own
    // experimental meta architecture, populate a key in the
    // model_builder.EXPERIMENTAL_META_ARCH_BUILDER_MAP dict and set its
    // value to a function that builds your model.
    ExperimentalModel experimental_model = 3;

    CenterNet center_net = 4;

    DETR detr = 5;
  }
}

message ExperimentalModel {
  optional string name = 1;
}
